### PR TITLE
Add a (crude) way to have structs in our UBOs

### PR DESCRIPTION
**Pull request by pixelflinger**
The struct must be declared in common_type.fs, so custom
structures are not supported.

### CODE REVIEW

File: shaders/src/common_types.fs [hunk 1]
```diff
@@ -22,3 +22,10 @@
 
 #define float3x3 mat3
 #define float4x4 mat4
+
+struct ShadowData {
+    highp mat4 spotlightFromWorldMatrix;
```
Comment (romainguy) [inline shaders/src/common_types.fs:27]: This won't work on Adreno, `highp` is ignored inside `struct`...

Comment (romainguy) [reply shaders/src/common_types.fs:27]: Although I guess that will work inside a UBO? If so you should probable leave a comment explaining this struct MUST be used in a UBO.

**Review by romainguy — APPROVED** [review]

File: libs/filabridge/include/private/filament/UniformInterfaceBlock.h [hunk 2]
```diff
@@ -63,23 +63,38 @@ class UniformInterfaceBlock {
...
+            return add(utils::StaticString{ uniformName }, size,
+                    utils::StaticString{ structName }, stride);
+        }
+
         // build and return the UniformInterfaceBlock
         UniformInterfaceBlock build();
     private:
         friend class UniformInterfaceBlock;
         struct Entry {
             Entry(utils::CString name, uint32_t size, Type type, Precision precision) noexcept
```
Comment (bejado) [inline libs/filabridge/include/private/filament/UniformInterfaceBlock.h:88]: Let's initialize `stride` here for non-struct types with `strideForType(type)`. That way `stride` is always accurate.

File: libs/filabridge/src/UniformInterfaceBlock.cpp [hunk 3]
```diff
@@ -89,7 +95,7 @@ UniformInterfaceBlock::UniformInterfaceBlock(Builder const& builder) noexcept
     uint16_t offset = 0;
     for (auto const& e : builder.mEntries) {
         size_t alignment = baseAlignmentForType(e.type);
-        uint8_t stride = strideForType(e.type);
+        uint8_t stride = strideForType(e.type, e.stride);
```
Comment (bejado) [inline libs/filabridge/src/UniformInterfaceBlock.cpp:98]: You should be able to simply do `e.stride` now.

**Pull request merged**